griptape/drivers/prompt/openai_chat_prompt_driver.py

```diff
@@ -90,3 +138,22 @@ def __to_openai_role(self, prompt_input: PromptStack.Input) -> str:
             return "assistant"
         else:
             return "user"
+
+    def _extract_ratelimit_metadata(self, response, *args, **kwargs):
+        # The timeparse utility doesn't handle sub-second durations as are sometimes returned by OpenAI's API.
+        # If the API returns, for example, "13ms", timeparse returns None. In this case, we will set the time value
+        # to the current time plus a one second buffer.
+        reset_requests_duration_sec = timeparse(response.headers["x-ratelimit-reset-requests"])
```

Comment:
Replace with:
`dateparser.parse(response.headers["x-ratelimit-reset-requests"], settings={'PREFER_DATES_FROM': 'future'})` 

This has the similar problem in that it will return None for something small like `13ms`.